src/spyglass/lfp/v1/analytic_signal.py

```diff
@@ -0,0 +1,129 @@
+from spyglass.common.common_interval import IntervalList
+from spyglass.common.common_nwbfile import AnalysisNwbfile
+
+schema = dj.schema("lfp_v1_analytic_signal")
+
+
+@schema
+class AnalyticSignalParameters(dj.Manual):
+    definition = """
+    analytic_signal_params_name: varchar(100) # name of the parameter set for detecting analytic signals
+    ---
+    analytic_signal_params: blob # a dict of analytic-signal detection parameters(the method name, e.g., hilbert transform, and specific parameters if any)
+    """
+
+    def insert_default(self):
+        """Insert the default artifact parameters with an appropriate parameter dict."""
+        analytic_signal_params = dict()
+        analytic_signal_params["analytic_method_name"] = "Hilbert transform"
+        analytic_signal_params[
+            "other_params"
+        ] = ""  # Hilbert transform doesn't contain other parameters; alternative methods may contain additional params.
+        self.insert1(["default", analytic_signal_params], skip_duplicates=True)
+
+
+@schema
+class AnalyticSignalSelection(dj.Manual):
+    definition = """
+    # Table that combines the LFP data, and parameters for generating analytic signals
+    -> LFPBand
+    -> IntervalList
+    -> LFPElectrodeGroup
+    -> AnalyticSignalParameters
+    ---
+    """
+
+
+@schema
+class AnalyticSignal(dj.Computed):
+    definition = """
+    -> AnalyticSignalSelection
+    ---
+    -> AnalysisNwbfile # the name of the nwb file with the analytic sinal & time stamps data
+    analytic_signal_object_id: varchar(40)  # the NWB object ID for loading this object from the file
+    """
+
+    def make(self, key):
+        # First, get the filtered lfp data
+        filtered_band = (LFPBand() & key).fetch_nwb()[0]["filtered_data"]
+
+        # Next, select data using defined electrode id(s) and defined valid interval.
+        electrode_list = (LFPElectrodeGroup.LFPElectrode() & key).fetch("electrode_id")
+        electrode_index = np.isin(filtered_band.electrodes.data[:], electrode_list)
+        start = (IntervalList() & key).fetch1("valid_times")[0][0]
+        end = (IntervalList() & key).fetch1("valid_times")[0][-1]
+        timestamps_selected = filtered_band.timestamps[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ]
+        filtered_band_selected = filtered_band.data[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ][:, electrode_index]
+
+        # Get analytic signal with the specified method (we're only using Hilbert transform for now)
+        filtered_analytic = np.empty(
+            shape=(filtered_band_selected.shape[0], 0), dtype=np.complex_
+        )
+        for lfp_electrode_index in range(filtered_band_selected.shape[1]):
+            filtered_lfp_one_channel = np.array(filtered_band_selected)[
```

Comment:
Is there a reason to assign this to `filtered_lfp_one_channel` instead of just running hilbert on the np.array you're creating:

```python
analytic_signal_one_channel = hilbert(
    np.array(filtered_band_selected)[:, lfp_electrode_index],
    axis=0
)
```